test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -1277,6 +1277,12 @@ public static Settings.Builder settings(Version version) {
         return Settings.builder().put(IndexMetadata.SETTING_VERSION_CREATED, version);
     }
 
+    /** Return consistent index settings for the provided index version, shard- and replica-count. */
+    public static Settings.Builder settings(Version version, int shards, int replicas) {
```

Comment:
This method name is pretty ambiguous, and even more difficult to read since the result is passed to a method named `settings`.

Could an alternative be to add a method to IndexMetadataBuilder that takes these 3 values?

Reply:
Admittedly the name sucks. I just found it equally weird to name it something different than what we already have with the existing method that I overloaded?
I figured, I'd just go on top of the existing thing, maybe dry things up a few rounds and then rename to something nicer when we ? Not sure we need this in prod code, there's really almost no places where we generate these settings?

Reply:
++ renamed :)